### PR TITLE
For #2346. Enable kotlin warningsAsErrors for `support-test` module.

### PR DESCRIPTION
**Pull request by dector**
### Issue #2346

  - Trivial changes in tests.
  - `ui-tabcounter` was fixed in #3073 but probably lost on merge.
  - Deprecated `createInstance()` as it has misleading name.

### Complexity

Trivial (★☆☆)

### Pull Request checklist
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] ~~**Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features~~

### CODE REVIEW

**Comment by dector** [comment]
Some tasks are failing after rebasing with latests master. Will fix 'em.

File: components/support/test/src/main/java/mozilla/components/support/test/KArgumentCaptor.kt [hunk 1]
```diff
@@ -32,11 +32,12 @@ class KArgumentCaptor<out T : Any?>(
 
     @Suppress("UNCHECKED_CAST")
     fun capture(): T {
-        return captor.capture() ?: createInstance(tClass) as T
+        return captor.capture() ?: castNull()
     }
 }
 
-fun <T : Any> createInstance(kClass: KClass<T>): T {
+@Deprecated("Try `mock()` or create instance manually")
```
Comment (jonalmeida) [inline components/support/test/src/main/java/mozilla/components/support/test/KArgumentCaptor.kt]: I think it's also safe to just remove this seeing as there aren't [any other usages](https://searchfox.org/mozilla-mobile/search?path=&q=createInstance).

Comment (dector) [reply components/support/test/src/main/java/mozilla/components/support/test/KArgumentCaptor.kt]: I agree. Just used deprecation to make transition smoother for external clients (if there are any). :)

Will remove it.

Comment (dector) [reply components/support/test/src/main/java/mozilla/components/support/test/KArgumentCaptor.kt]: @jonalmeida Done.

**Pull request merged**